internal/providers/node_runtime_deps_info.bzl

```diff
@@ -70,6 +70,8 @@ def run_node(ctx, inputs, arguments, executable, chdir = None, **kwargs):
         arguments: list or ctx.actions.Args object containing arguments to pass to the executable
         executable: stringy representation of the executable this action will run, eg eg. "my_executable" rather than ctx.executable.my_executable
         chdir: directory we should change to be the working dir
+        stamp_env_vars: if True ctx.version_file and ctx.info_file are supplied as inputs to the action
```

Comment:
I think the doc here needs to indicate plus and minus of doing this, like

"Allows the build result to contain version metadata like the git SHA, but makes the result non-determistic which causes cache misses for anything depending on it"

Also, I'm surprised this doesn't seem to be conditional on whether the build is performed with `--stamp` which I think it should be

Reply:
The condition check is done where `run_node` is called. esbuild.bzl & npm_package_bin both have
```
stamp_env_vars = ctx.attr.node_context_data[NodeContextInfo].stamp,
```

Suppose the condition could be moved to within `run_node` itself if the convention for the conditional attribute was always `node_context_data`

Reply:
Added that warning to the docstring

Reply:
Refactored so run_node sniffs out the "stamp" attr